language/move-compiler/src/naming/translate.rs

```diff
@@ -507,15 +560,37 @@ fn function(
         body,
     };
     fake_natives::function(context.env, module_opt, name, &f);
+    let used_locals = std::mem::take(&mut context.used_locals);
+    remove_unused_bindings_function(context, &used_locals, &mut f);
+    context.local_scopes = vec![];
```

Comment:
There's a potential risk that some function-specific field could get added and then not reset here -- is it worth grouping these fields into a single struct, which you `std::mem::take` at the end here, to guarantee that they all get reset?

Reply:
> is it worth grouping these fields into a single struct,

Yes... and I've been really bad about this throughout the compiler. I might want to fix this everywhere. Maybe another PR?

Reply:
Another PR sounds good to me.